api/src/test/java/org/openmrs/module/fhir2/providers/r4/ConditionFhirResourceProviderTest.java

```diff
@@ -213,4 +235,32 @@ public void searchConditions_shouldReturnConditionReturnedByServiceWhenPatientIs
 		assertThat(resultList.size(), greaterThanOrEqualTo(1));
 		assertThat(resultList.iterator().next().fhirType(), equalTo(FhirConstants.CONDITION));
 	}
+	
+	@Test(expected = MethodNotAllowedException.class)
+	public void updateConditionShouldThrowMethodNotAllowedIfDoesNotExist() {
+		
+		Condition wrongCondition = new Condition();
+		wrongCondition.setId(WRONG_CONDITION_UUID);
+		
+		when(conditionService.update(eq(WRONG_CONDITION_UUID), any(Condition.class)))
+		        .thenThrow(MethodNotAllowedException.class);
+		
+		resourceProvider.updateCondition(new IdType().setValue(WRONG_CONDITION_UUID), wrongCondition);
+		verify(conditionService.update(eq(WRONG_CONDITION_UUID), any(Condition.class)));
+		
```

Comment:
I think you have to have the same set of unit tests here as well @sherrif10  as r3

Reply:
sorry for late reply however let me bump into this tomorow thanks